src/dft/src/Utils.hh

```diff
@@ -52,4 +54,38 @@ odb::dbInst* ReplaceCell(
 // otherwise
 bool IsSequentialCell(sta::dbNetwork* db_network, odb::dbInst* instance);
 
+// Returns a vector of dbITerm for every clock that there is in the instance.
+// For black boxes or CTLs, we can have more than one clock
+std::vector<odb::dbITerm*> GetClockPin(odb::dbInst* inst);
+
+// Returns a sta::Clock of the given iterm
+std::optional<sta::Clock*> GetClock(sta::dbSta* sta, odb::dbITerm* iterm);
+
+// Runs the given function in a child process. We can use this function as a way
+// to rollback changes to the netlist
+void RunInForkForRollback(const std::function<void()>& fn);
```

Comment:
How will this work if you are in the GUI and the logger's output is being directed into the GUI's output pane?

Reply:
We don't have a generic roll back today.  rsz built a local one but it only works for the specific sorts of changes it makes (buffering/resizing).

As this won't work in this formulation I don't think we should include it.  You could make a blunt tool like delete all scan chains and start again.

Reply:
I think I can do something for scan replace where I keep track of the original cell and replace back the cell after preview. I will try to implement something like that and ping you once it is ready for review again. Thanks!